internal/exec/stages/mount/mount.go

```diff
@@ -142,31 +142,35 @@ func (s stage) mountFs(fs types.Filesystem) error {
 		return err
 	}
 
-	if distro.SelinuxRelabel() {
-		if err := s.RelabelFiles([]string{firstMissing}); err != nil {
-			return err
+	isReadOnly := util.FileSystemIsReadOnly(path)
```

Comment:
This seems like a pretty indirect way to go about things.  If the filesystem is read-only and we needed to create the mountpoint directory, we already failed to do that before we got here.  If the mountpoint directory already existed, we'll relabel the mountpoint _most of the time_, but silently skip doing so if the filesystem is read-only.  That special case feels wrong - it could surprise a user who's expecting their mountpoint to be relabeled - so it's worth stepping back and looking for a different way to solve this.

The problem as described in https://github.com/coreos/ignition/issues/1452#issuecomment-1224386412 is that `FindFirstMissingPathComponent` doesn't distinguish whether the specified path already exists or not.  If the directory already exists, maybe we don't need to relabel it at all?

[Blame](https://github.com/coreos/ignition/blame/24c18840349aa9b981055e12f2ec028b3e8a609f/internal/exec/stages/mount/mount.go#L145) points to 381132ff15f30744b6af2a05fea85e3d0ca1cb5b, whose commit message says it wants to relabel `the mount point we create`.  And the code from that commit calls a function named `FindFirstMissingDirForFile`, which is consistent with the idea that it wants to relabel a _newly-created_ directory.  But that's not what the function is named now.  Looking at [blame](https://github.com/coreos/ignition/blame/24c18840349aa9b981055e12f2ec028b3e8a609f/internal/exec/stages/mount/mount.go#L129) again, we find the function rename in e30dfa4add1b0ffe8aef7ce0ddd1721489ba07f6.  And sure enough, that commit clarifies the function's behavior if all directory components exist.  So it seems like the original commit was based on an incomplete analysis of the "already exists" case.

On the basis of that record, I think the right approach is to relabel only if the directory is newly created.  Since creating the directory can't succeed on a read-only filesystem, that'll solve #1452.  I think @jlebon, the author of 381132ff15f30744b6af2a05fea85e3d0ca1cb5b, will agree with that analysis, since he retitled #1452 to `Don't try to relabel mountpoints which already exist` last August. :slightly_smiling_face:

We're already checking whether the mountpoint exists on line 135, so this should be a fairly small change to the original code.

Reply:
Okay, so correct me if i'm wrong but it seems like i just need to write a function to check if the specified directory exists and if it does exist do nothing?

Reply:
As I said, we're already doing this on line 135, so we should just be able to move some code under that conditional.

Reply:
okay that makes sense, and how long should a directory be considered newly created?
should it be considered "old" if its existed longer than x amount of minutes?

Reply:
"Newly created" means we just created it.  If Ignition creates the directory, we should relabel it, and if the directory is already there, we shouldn't.